sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -75,7 +77,7 @@ class FileStreamSource(
 
   /** A mapping from a file that we have processed to some timestamp it was last modified. */
   // Visible for testing and debugging in production.
-  val seenFiles = new SeenFilesMap(sourceOptions.maxFileAgeMs)
+  val seenFiles = new SeenFilesMap(sourceOptions.maxFileAgeMs, sourceOptions.fileNameOnly)
```

Comment:
It's better to add a warning when `fileNameOnly` is true. How about
```
logWarning("fileNameOnly is enabled. Make user your file names are unique (e.g., using UUID), otherwise, files using the same name will be considered as the same file and causes data lost")
```